### PR TITLE
docs: fix assorted typos

### PR DESCRIPTION
**Pull request by nnsW3**
This PR addresses several typographical errors across various files in the project. The changes improve readability and maintain the professional standard of the documentation and code comments.

Justification
Typographical errors, while minor, can detract from the overall quality of the project. Correcting these errors ensures clarity and professionalism, making the project more accessible and understandable for current and future contributors.

### CODE REVIEW

File: docs/concepts/composable-ecosystem.qmd [hunk 1]
```diff
@@ -3,7 +3,7 @@
 Ibis exists in a broader composable data ecosystem. [The Composable Codex by
 Voltron Data](https://voltrondata.com/codex) is the result of years of
 experience and hours of writing by experts in the field, providing an in-depth
-introduction the composable data systems. We'll take a look at:
+introduction of composable data systems. We'll take a look at:
```
Comment (cpcloud) [inline docs/concepts/composable-ecosystem.qmd:6]:
```suggestion
introduction to composable data systems. We'll take a look at:
```

File: docs/how-to/configure/basics.qmd [hunk 2]
```diff
@@ -13,7 +13,7 @@ t = ibis.examples.penguins.fetch()  # <2>
 
 ## Overview
 
-Ibis configuration happens through the `ibis.options` attribute. Attributes can be get and set like class attributes.
+Ibis configuration happens through the `ibis.options` attribute. Attributes can be got and set like class attributes.
```
Comment (cpcloud) [inline docs/how-to/configure/basics.qmd:16]:
```suggestion
Ibis configuration happens through the `ibis.options` attribute. Attributes can be read and written like class attributes.
```

File: docs/backends/impala.qmd [hunk 3]
```diff
@@ -373,7 +373,7 @@ render_methods(
 ```
 
 To address a specific partition in any method that is partition
-specific, you can either use a dict with the partition key names and
+Specifically, you can either use a dict with the partition key names and
```
Comment (cpcloud) [inline docs/backends/impala.qmd:376]: This is correct as it was, but perhaps changing `that is partition specific` to `that is partition-specific` would make it more clear?

**Review by cpcloud — CHANGES_REQUESTED** [review]

**Comment by nnsW3** [comment]
done

**Comment by cpcloud** [comment]
@nnsW3 By any chance did you use an LLM in any part of this process?

**Pull request closed without merging**